bin/centreon-sanitize-images.php

```diff
@@ -0,0 +1,315 @@
+ * Linking this program statically or dynamically with other modules is making a
+ * combined work based on this program. Thus, the terms and conditions of the GNU
+ * General Public License cover the whole combination.
+ *
+ * As a special exception, the copyright holders of this program give Centreon
+ * permission to link this program with independent modules to produce an executable,
+ * regardless of the license terms of these independent modules, and to copy and
+ * distribute the resulting executable under terms of Centreon choice, provided that
+ * Centreon also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ * SVN : $URL$
+ * SVN : $Id$
+ *
+ */
+require(__DIR__ . '/../vendor/autoload.php');
+use enshrined\svgSanitize\Sanitizer;
+
+// Get Script options
+$options = getopt('h', ['help::','list', 'sanitize']);
+
+if (empty($options)) {
+    echo "Missing parameters --list or --sanitize";
+    exit(1);
+}
+
+if (isset($options['help']) || isset($options['h'])) {
+    usage();
+    exit(0);
+}
+
+// --list option
+if (isset($options['list'])) {
+    $files = listImages();
+    if (empty($files['svgImages']) && empty($files['invalidFiles'])) {
+        echo PHP_EOL . "Nothing to do, everything is fine." . PHP_EOL;
+    } else {
+        echo PHP_EOL . "You can execute the script with the --sanitize options to apply the modifications." . PHP_EOL;
+    }
+    exit(0);
+}
+
+// --sanitize option
+if (isset($options['sanitize'])) {
+    $files = listImages();
+    if (empty($files['svgImages']) && empty($files['invalidFiles'])) {
+        echo PHP_EOL . "Nothing to do, everything is fine." . PHP_EOL;
+    }else {
+        if (!empty($files['svgImages'])) {
+            foreach($files['svgImages'] as $svgImage) {
+                sanitizeSvg($svgImage);
+            }
+        }
+        if (!empty($files['invalidFiles'])) {
+            foreach ($files['invalidFiles'] as $invalidImg) {
+                convertCorruptedImage($invalidImg);
+            }
+        }
+        echo "Sanitize done." . PHP_EOL;
+        exit(0);
+    }
+}
+
+###############################
+#        COMMON FUNCTION      #
+###############################
+
+/**
+ * Display usage message.
+ *
+ */
+function usage(): void
+{
+    echo <<<'EOD'
+
+This script will sanitize all your svg files and replace your corrupted images by a generic image.
+
+Global Options:
+    --list : Will list all your svg files and/or your corrupted images
+    --sanitize : Will list all your svg files and/or your corrupted images and sanitize/replace them.
+    -h,--help: display this message.
+
+
```

Comment:
```suggestion
```

Reply:
it is necessary to properly display the next line:

```
[root@localhost bin]# php centreon-sanitize-images.php --help

This script will sanitize all your svg files and replace your corrupted images by a generic image.   

Global Options:
    --list : Will list all your svg files and/or your corrupted images
    --sanitize : Will list all your svg files and/or your corrupted images and sanitize/replace them.
    -h,--help: display this message.

[root@localhost bin]# 
```

Reply:
"/n" could have done it ;) 
next dev working on it may remove this space

Reply:
not sur \n is correctly interpreted in a EOD

Reply:
I'm not sure /n is correctly interpreted into a EOD